### DISCUSSION TITLE
Strange behavior animation position

### DISCUSSION THREAD

**Discussion by moiatto** [mapsforge/vtm#1112] [Q&A] [answered]
I created this function that updates my location on the map when I receive GPS data (FusedLocationProviderClient) approximately every +/- 1 second. 
It is the only function that uses animations:

```
    private void updateUIwithMapLocationAnimation(MapPosition position)
    {
        map.animator().animateTo(980, position);   // ms

        map.updateMap(true);
        map.viewport().syncViewport();
    }
```

**My issue:**
The animation is fluid for some time (even several minutes), then it appears not fluid as if it were losing frames. This always happens and with variable times. It can happen after 10 seconds or minutes.

_What I noticed is that if I tap the map, it returns to fluid immediately and then starts again._

I tried inserting a `map.animator().cancel()` before the animation but it doesn't change behavior.

**Comment by devemux86** [comment]
Try the `animateTo` without the other code.

**Reply by moiatto** [reply, marked as answer]
Hi Emux,
I solved thanks to your suggestions:
```
private void updateUIwithMapLocationAnimation(MapPosition position)
    {
        map.animator().cancel();
        
        map.animator().animateTo(1000, position);
    }
```

**Comment by devemux86** [comment]
Smooth animation can be done if its duration is similar to GPS location updates.
If using 1 sec animation and GPS location is updated every 1.5 sec then it won't be smooth.

In such cases a Kalman filter could help.

**Comment by moiatto** [comment]
Thanks Emux.

I set the GPS location update request to 1 second.
I will try your suggestions.

For the Kalman Filter can you tell me how to implement it because I have no idea.
